pkg/network/ebpf/c/protocols/postgres/decoding.h

```diff
@@ -240,20 +283,33 @@ static __always_inline bool handle_response(pktbuf_t pkt, conn_tuple_t conn_tupl
         // the message tag. So we need to add 1 to the message length to jump over the entire message.
         pktbuf_advance(pkt, header.message_len + 1);
     }
+    iteration_value->total_msg_count += messages_count;
 
     if (found_command_complete) {
+        handle_command_complete(&conn_tuple, transaction);
+        update_msg_count_telemetry(pg_msg_counts, iteration_value->total_msg_count);
```

Comment:
what is the value of this PR if you update the telemetry only if we've found a command complete?

Reply:
and what is a packet has only 80 messages in it, without having `command complete`?

Reply:
The accuracy of the Postgres handler depends on its ability to detect Postgres completion commands. When reading a TCP packet, the following outcomes are possible:
1. A completion command is found.
2. The packet does not contain a completion command.
3. The packet is fragmented.
4. The maximum message counter is reached.
Cases 3 and 4 reduce accuracy, and their impact can be measured by the relative number of failed cases. In my initial implementation, I compared cases 3 and 4 against case 1, ignoring case 2 as irrelevant. However, based on my understanding of your previous comment, you disagreed with this approach. I have since updated the code to also count case 2.
To address your question: with the current implementation, a packet containing 80 messages without a completion command would be counted in the first bucket.
Ultimately, it involves a human interpretation of the results and an evaluation of the impact caused by the missed completion commands.

Reply:
it will be counted by field `pg_msg_counts->reached max_messages` as it exceeds allowed maximum. Whenever outcome is 3 or 4 we cannot say if completion command was in packet or was not.

Reply:
Addition. All possible outcomes handled on following lines. Line 290 adds to bucket when discovered completion command, Line 297 saves counter when reached maximum of messages, line 302 covers case of end of packet, line 307 adds to counter of broken packet.